### ISSUE TITLE
RFC0593: suggest we make `options` a JSONLD object such that it can be extended

### ISSUE THREAD

**Issue by llorllale** [hyperledger/aries-rfcs#662]
The [`options`](https://github.com/hyperledger/aries-rfcs/blob/master/features/0593-json-ld-cred-attach/README.md#ld-proof-vc-detail-attachment-format) object of the `aries/ld-proof-vc-detail@v1.0` format is a plain old JSON object. It would be better if this object is actually a JSONLD document with `@context` and `type` such that it can be extended by different implementations. Of course, the definition of a core Aries JSONLD options type would still be useful and part of the JSONLD profile subtarget for interop.

The primary motivation for this is we would like enable the use of P curves with this attachment format, and we'd like to use `JsonWebSignature2020` for it.

Note: specifying the verification method's type (`JsonWebKey2020`) is insufficient as the underlying key could be anything. We'd like to be able to specify the exact P curve to use (P256, P384, P512).

cc @TimoGlastra

**Comment by TimoGlastra** [comment]
I agree. As we're already in JSON-LD land it makes sense to utilize its features

How is DIF PE specify the key type? Seems like it is also missing this. Or is it expected that you support all underlying key types if you support the proof type?

Maybe we should have taken the same route as the VC HTTP API where you specify the `verificationMethod` instead of the `proofType`. Which is sufficient to determine the key type.
